### PR TITLE
UI layout tree debug print

### PR DESCRIPTION
**Pull request by ickshonpe**
# Objective

Copy the `debug::print_tree` function from Taffy except display entity ids instead of Taffy's node ids and indicate which ui nodes have a measure func.

### CODE REVIEW

File: crates/bevy_ui/src/layout/debug.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
+use crate::UiSurface;
+use bevy_ecs::prelude::Entity;
+use bevy_utils::HashMap;
+use taffy::prelude::Node;
+use taffy::tree::LayoutTree;
+
+pub fn print_ui_layout_tree(ui_surface: &UiSurface) {
```
Comment (alice-i-cecile) [inline crates/bevy_ui/src/layout/debug.rs:7]: Basic docs please.

Comment (alice-i-cecile) [reply crates/bevy_ui/src/layout/debug.rs:7]: Actually: why isn't this just a `Display` impl?

Comment (ickshonpe) [reply crates/bevy_ui/src/layout/debug.rs:8]: I'm not sure what the conventions about `Display` impls are. This doesn't print everything known by `UiSurface`, only the layout tree.

**Review by alice-i-cecile — APPROVED** [review]

**Review by mockersf — CHANGES_REQUESTED** [review]
I would prefer to avoid `println!` and use one of the log macro as they are supported on all platforms

**Comment by ickshonpe** [comment]
> I would prefer to avoid `println!` and use one of the log macro as they are supported on all platforms

`info!` seems like the right level?

**Review by nicoburns — APPROVED** [review]
This seems fine. Seems like ideally Taffy's function would be made more generic. But probably easier to just do it like this for now.

**Comment by ickshonpe** [comment]
> This seems fine. Seems like ideally Taffy's function would be made more generic. But probably easier to just do it like this for now.

Yeah, this is just a starting point, as our debugging options are so poor atm. Maybe a version that uses a `SystemParam` to access the style and content data so that it can walk the tree and point out potential problems could be useful as well.

**Pull request merged**